contents/tutorials/track-high-volume-apis.md

```diff
@@ -0,0 +1,278 @@
+
+```node
+//...
+const eventSampling = await client.isFeatureEnabled(
+	'sample-flag',
+	req.params.id,
+	{ sendFeatureFlagEvents: false }
+);
+//...
+```
+
+### Locally evaluating our flag
+
+Another problem is that every feature flag call on the server side sends a request to PostHog to evaluate the flag. In a high volume API endpoint, the time it takes to make this request can add up. We can limit this by evaluating the flag locally using the person properties the flag relies on.
+
+Because we initialize the `posthog-node` SDK with our personal API key, setting this up is simple. Just add `firstRouteCalled` in `personProperties` to the options in our `client.isFeatureEnabled()` call.
+
+```node
+//...
+const eventSampling = await client.isFeatureEnabled(
+    'sample-flag',
+    req.params.id,
+    { 
+      sendFeatureFlagEvents: false,
+      personProperties: {
+        'firstRouteCalled': req.url,
+      },
+    }
+  );
+//...
+```
+
+In our small example, this reduced response time by about 100ms. A massive amount for high-volume APIs where every millisecond counts.
+
+## Option 2. Cache usage and batch events
+
+Another way to capture events for high-volume APIs is to cache usage of the API and then capture a batch of events at regular intervals. 
+
+To do this, we use [node-cache](https://www.npmjs.com/package/node-cache) to create a basic cache to store the number of requests, then use [node-cron](https://www.npmjs.com/package/node-cron) to set up a cron job to read the cache, capture values for each of the IDs, and clear the cache. 
+
+First, install them both.
+
+```bash
+npm i node-cache node-cron
+```
+
+Next, set up our cache. It simply gets the amount of API calls for a param ID and then increments it.
+
+```node
+// index.js
+//...
+const NodeCache = require('node-cache');
+const cache = new NodeCache()
+
+//...
+
+app.get('/big/:id', async (req, res) => {
+  let apiCalls = cache.get(req.params.id) || 0;
+  apiCalls++;
+  cache.set(req.params.id, apiCalls);
+
+  res.send(`Welcome to the big route, ${req.params.id}`);
+})
+//...
+```
+
+We can now set up our cron job to loop through the keys in the cache, get the number of events for that key, capture a batch event with a distinct ID (we use later), and then clear the cache once completed.
+
+```node
+// index.js
+//...
+const cron = require('node-cron');
+
+//...
+
+cron.schedule('*/10 * * * * *', () => {
+  cache.keys().forEach((key) => {
+    let apiCalls = cache.get(key);
+    client.capture(
+      { 
+        event: 'big route batch',
+        distinctId: key,
+        properties: { 
+          calls: apiCalls,
+	  distinctId: key
+        },
+      },
+    );
+  });
+  cache.flushAll();
+});
+```
+
+Once you restart the server and send multiple events to different IDs, you will see the event batches in your PostHog instance. Each batch event has the amount of API calls as a property which we can use for our analysis.
```

Comment:
This sentence is confusing. What do you mean by restart the server? Wouldn't that lose all state?
Do you mean "flush the cache" ?

Reply:
You need to restart the server to implement the changes, I'll clarify.